src/loaders/compute_tools/microtrait/microtrait_util.py

```diff
@@ -48,21 +107,14 @@ def create_trait_unwrapped_rules(
         for rule_name in rule_names.copy():
             unwrapped_rule = rule_dict.get(rule_name)
             if unwrapped_rule:
-                # find and extract gene names from the unwrapped rule expression.
-                # i.e. "('pcbA' | 'pcbB' | 'pcbC' | 'pcbD' | 'pcbE' | 'pcbF' | 'pcbG' | 'pcbH')"
-                gene_names = set(re.findall(r"'(\w+)'", unwrapped_rule))
+                gene_names = _retrieve_genes(unwrapped_rule)
                 # remove the rule name from the set and add the unwrapped rule expression
                 trait_rule_unwrapped_mapping[trait_name].remove(rule_name)
                 trait_rule_unwrapped_mapping[trait_name].update(gene_names)
             else:
                 raise ValueError(f"Rule name {rule_name} not found in ruleunwrapped file")
 
```

Comment:
Can you add some sort of check to ensure all traits have rules (I assume it's possible that some traits may have no genes in the gene list, but they should all have rules) if there isn't one already?

Reply:
👍

Reply:
:+1: